packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts

```diff
@@ -369,15 +374,20 @@ export class OdspSummaryUploadManager {
                                 encoding: "base64",
                             };
                         }
-                        hash = await hashFile(IsoBuffer.from(value.content, value.encoding));
-                        cachedPath = this.blobTreeDedupCaches.blobShaToPath.get(hash);
+                        if (this.hostPolicy.blobDeduping) {
+                            hash = await hashFile(IsoBuffer.from(value.content, value.encoding));
+                            cachedPath = this.blobTreeDedupCaches.blobShaToPath.get(hash);
+                        }
                     }
                     (summaryObject as any).content = undefined;
                     // If the cache has the hash of the blob and handle of last summary is also present, then use that
                     // cached path for the given blob. Also update the caches for future use.
                     if (cachedPath === undefined || parentHandle === undefined) {
                         blobs++;
                     } else {
+                        if (!this.hostPolicy.blobDeduping) {
```

Comment:
Not sure here. If this is the case we should never get to, then it should be an assert. If it's normal path to get to, then it should not be an error event

Reply:
Added assert. Should never come here when blob deduping is disabled.